format-common/src/main/java/co/cask/hydrator/format/input/TextInputProvider.java

```diff
@@ -32,14 +32,13 @@
   public FileInputFormatter create(Map<String, String> properties, @Nullable Schema schema) {
     if (schema != null) {
       Schema.Field offsetField = schema.getField("offset");
-      if (offsetField == null) {
-        throw new IllegalArgumentException("Schema for text format must have a field named 'offset'");
-      }
-      Schema offsetSchema = offsetField.getSchema();
-      Schema.Type offsetType = offsetSchema.isNullable() ? offsetSchema.getNonNullable().getType() :
-        offsetSchema.getType();
-      if (offsetType != Schema.Type.LONG) {
-        throw new IllegalArgumentException("Type of 'offset' field must be 'long', but found " + offsetType);
+      if (offsetField != null) {
+        Schema offsetSchema = offsetField.getSchema();
+        Schema.Type offsetType = offsetSchema.isNullable() ? offsetSchema.getNonNullable().getType() :
+          offsetSchema.getType();
+        if (offsetType != Schema.Type.LONG) {
+          throw new IllegalArgumentException("Type of 'offset' field must be 'long', but found " + offsetType);
+        }
```

Comment:
Do we also need changes near line 63?

Reply:
no that just gets the default schema. Renaming that method.